components/earnings/commons/earnings_base.lua

```diff
@@ -123,75 +141,45 @@ end
 -- @year - (optional) the year to calculate earnings for
 -- @mode - (optional) the mode to calculate earnings for
 -- @perYear - (optional) query all earnings per year and return the values in a lua table
--- @divisionFactor - divisionFactor function
----
--- customizable in case query has to be changed
--- (e.g. SC2 due to not having a fixed number of players per team)
-function Earnings.calculate(conditions, year, mode, perYear, divisionFactor)
-	conditions = Earnings._buildConditions(conditions, year, mode)
-
-	if Logic.readBool(perYear) then
-		return Earnings.calculatePerYear(conditions, divisionFactor)
-	end
-
-	local prizePoolColumn = Earnings._getPrizePoolType(divisionFactor)
-
-	local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
-		conditions = conditions,
-		query = 'mode, sum::' .. prizePoolColumn,
-		groupby = 'mode asc'
-	})
+-- @aliases - players/teams to determine earnings for
+function Earnings.calculate(conditions, queryYear, mode, perYear, aliases, isPlayerQuery)
+	conditions = Earnings._buildConditions(conditions, queryYear, mode)
 
+	local sums = {}
 	local totalEarnings = 0
+	local sumUp = function(placement)
+		local value = Earnings._determineValue(placement, aliases, isPlayerQuery)
+		if perYear then
+			local year = string.sub(placement.date, 1, 4)
+			if not sums[year] then
+				sums[year] = 0
+			end
+			sums[year] = sums[year] + value
+		end
 
-	for _, item in ipairs(lpdbQueryData) do
-		local prizeMoney = item['sum_' .. prizePoolColumn]
-		totalEarnings = totalEarnings + Earnings._applyDivisionFactor(prizeMoney, divisionFactor, item['mode'])
+		totalEarnings = totalEarnings + value
 	end
 
-	return MathUtils._round(totalEarnings)
-end
+	local queryParameters = {
```

Comment:
```suggestion
	totalEarnings = MathUtils._round(totalEarnings)

	local queryParameters = {
```

Reply:
needs to be below the `Lpdb.executeMassQuery` call!